dev-db/redict/redict-7.3.0.ebuild

```diff
@@ -0,0 +1,172 @@
+	)"
+
+PATCHES=(
+	"${FILESDIR}"/redict-6.2.1-config.patch
+	"${FILESDIR}"/redict-sentinel-7.2.0-config.patch
+	"${FILESDIR}"/redict-7.3.0-system-jemalloc.patch
+)
+
+src_prepare() {
+	default
+
+	# Respect user CFLAGS in bundled lua
+	sed -i '/LUA_CFLAGS/s: -O2::g' deps/Makefile || die
+	# The makefile rebuilds all binaries if
+	# arguments change between building and installing
+	sed -i '/^install:/s: all::g' src/Makefile || die
+}
+
+src_compile() {
+	tc-export AR CC RANLIB
+
+	local myconf=(
+		AR="${AR}"
+		CC="${CC}"
+		RANLIB="${RANLIB}"
+		CFLAGS="${CFLAGS}"
+		LDFLAGS="${LDFLAGS}"
+
+		V=1 # verbose
+
+		# OPTIMIZATION defaults to -O3. Let's respect user CFLAGS by setting it
+		# to empty value.
+		OPTIMIZATION=''
+		# Disable debug flags in redict
+		DEBUG=''
+		# Disable debug flags in bundled hiredis
+		DEBUG_FLAGS=''
+
+		BUILD_TLS=$(usex ssl)
+		USE_SYSTEMD=$(usex systemd)
+	)
+
+	if use jemalloc; then
+		myconf+=(
+			MALLOC=jemalloc
+			USE_SYSTEM_JEMALLOC=yes
+		)
+	elif use tcmalloc; then
+		myconf+=( MALLOC=tcmalloc )
+	else
+		myconf+=( MALLOC=libc )
+	fi
+
+	emake -C src "${myconf[@]}"
+}
+
+src_test() {
+	local runtestargs=(
+		--clients "$(makeopts_jobs)" # see bug #649868
+
+		--skiptest "Active defrag eval scripts" # see bug #851654
+	)
+
+	if has usersandbox ${FEATURES} || ! has userpriv ${FEATURES}; then
+		ewarn "oom-score-adj related tests will be skipped." \
+			"They are known to fail with FEATURES usersandbox or -userpriv. See bug #756382."
+
+		runtestargs+=(
+			# unit/oom-score-adj was introduced in version 6.2.0
+			--skipunit unit/oom-score-adj # see bug #756382
+
+			# Following test was added in version 7.0.0 to unit/introspection.
+			# It also tries to adjust OOM score.
+			--skiptest "CONFIG SET rollback on apply error"
+		)
+	fi
+
+	if use ssl; then
+		edo ./utils/gen-test-certs.sh
+		runtestargs+=( --tls )
+	fi
+
+	edo ./runtest "${runtestargs[@]}"
+}
+
+src_install() {
+	insinto /etc/redict
+	doins redict.conf sentinel.conf
+	use prefix || fowners -R redis:redis /etc/redict /etc/redict/{redict,sentinel}.conf
+
+	newconfd "${FILESDIR}/redict.confd" redict
+	newinitd "${FILESDIR}/redict.initd" redict
+
+	systemd_newunit "${FILESDIR}/redict.service" redict.service
+	newtmpfiles "${FILESDIR}/redict.tmpfiles" redict.conf
+
+	newconfd "${FILESDIR}/redict-sentinel.confd" redict-sentinel
+	newinitd "${FILESDIR}/redict-sentinel.initd" redict-sentinel
+
+	insinto /etc/logrotate.d/
+	newins "${FILESDIR}/redict.logrotate" "${PN}"
+
+	emake install DESTDIR="${ED}" PREFIX="/usr"
+
+	if use prefix; then
+		diropts -m0750
+	else
+		diropts -m0750 -o redis -g redis
+	fi
+	keepdir /var/{log,lib}/redict
+}
+
+pkg_postinst() {
+	tmpfiles_process redict.conf
+
+	if [[ -z "${REPLACING_VERSIONS}" ]]; then
+		# In case of a new installation
+		ewarn "The default redict configuration file location changed to:"
+		ewarn "  /etc/redict/{redict,sentinel}.conf"
+		ewarn "Please apply your changes to the new configuration files."
```

Comment:
I think this message will be confusing for those not having `dev-db/redis` installed. We should also check `has_version dev-db/redis` and maybe change the wording a bit, because we want to make it clear that this message applies for those migrating from redis.

Reply:
added the check for a redis version, tho i'm unsure what to change the wording to

Reply:
I would suggest something like:
> Redict uses different configuration files than redis:
>   /etc/redict/{redict,sentinel}.conf
> Please apply your changes to the new configuration files.

Reply:
that does sound nicer indeed, reworded to it